zmq/sugar/socket.py

```diff
@@ -120,6 +121,66 @@ def close(self, linger=None):
             self.context._rm_socket(self)
         super(Socket, self).close(linger=linger)
 
+    #-------------------------------------------------------------------------
+    # Connect/Bind context managers
+    #-------------------------------------------------------------------------
+
+    @contextmanager
+    def _connect_cm(self, addr):
+        """Context manager to disconnect on exit"""
+        try:
+            yield
+        finally:
+            self.disconnect(addr)
+
+    @contextmanager
+    def _bind_cm(self, addr):
+        """Context manager to unbind on exit"""
+        try:
+            yield
+        finally:
+            self.unbind(addr)
+
+    def bind(self, addr):
+        """s.bind(addr)
+
+        Bind the socket to an address.
+
+        This causes the socket to listen on a network port. Sockets on the
+        other side of this connection will use ``Socket.connect(addr)`` to
+        connect to this socket.
+
+        Returns a context manager that may be used to automatically unbind in a `with` block.
+
+        Parameters
+        ----------
+        addr : str
+            The address string. This has the form 'protocol://interface:port',
+            for example 'tcp://127.0.0.1:5555'. Protocols supported include
+            tcp, udp, pgm, epgm, inproc and ipc. If the address is unicode, it is
+            encoded to utf-8 first.
+        """
+        SocketBase.bind(self, addr)
```

Comment:
We can use super to call the parent method:

```suggestion
        super().bind(addr)
```

Reply:
Sure. I'm sometimes hesitant to use `super()` when there's multiple inheritance (possibly due to problems with my understanding of multiple inheritance), but it's probably safe here.